crates/miden-proving-service-client/Cargo.toml

```diff
@@ -32,7 +33,7 @@ tonic-web = { version = "0.12", optional = true }
 async-trait = "0.1"
 miden-objects = { workspace = true, default-features = false }
 miden-tx = { workspace = true, default-features = false }
-miden-tx-batch-prover = { workspace = true, default-features = false }
+miden-tx-batch-prover = { workspace = true, default-features = false, optional = true}
```

Comment:
nit: missing space before `}` - but also, why do we need the `miden-tx-batch-prover` dependency?

Also, shouldn't `miden-tx` be an optional dependency?

Reply:
we don't use it anymore, I was using the `BatchProverError` from there. I'm removing it.

Regarding `miden-tx`, I ended up moving the conversion between to request/responses to native types into each specialized module (`tx-prover.rs`, `batch-prover.rs` and `block-prover.rs`). With that I'm making miden-tx optional.